packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingPlugin.java

```diff
@@ -72,24 +63,14 @@ private void initInstance(BinaryMessenger messenger) {
     registerPlugin(channelName, this);
   }
 
-  private void onAttachedToEngine(Context context, BinaryMessenger binaryMessenger) {
-    initInstance(binaryMessenger);
-  }
-
-  private void setActivity(Activity flutterActivity) {
-    this.mainActivity = flutterActivity;
-  }
-
   @Override
   public void onAttachedToEngine(FlutterPluginBinding binding) {
-    onAttachedToEngine(binding.getApplicationContext(), binding.getBinaryMessenger());
+    initInstance(binding.getBinaryMessenger());
   }
 
   @Override
   public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
-    if (binding.getApplicationContext() != null) {
```

Comment:
Add the if statement back in please, we're working on the assumption that this might be a [solution](https://github.com/FirebaseExtended/flutterfire/pull/7088) to a crash experienced by some users.